tvloader/parser2v1/parse_file_test.go

```diff
@@ -885,4 +886,58 @@ func TestParser2_1FailsIfArtifactURIBeforeArtifactName(t *testing.T) {
 	}
 }
 
+func TestParser2_1FilesWithoutSpdxIdThrowError(t *testing.T) {
+	// case 1
+	// Last unpackaged file no packages in doc
+	// Last file of last package in the doc
+	tvPairs := []reader.TagValuePair{
+		{Tag: "SPDXVersion", Value: "SPDX-2.1"},
+		{Tag: "DataLicense", Value: "CC0-1.0"},
+		{Tag: "SPDXID", Value: "SPDXRef-DOCUMENT"},
+		{Tag: "FileName", Value: "f1"},
+	}
+	_, err := ParseTagValues(tvPairs)
+	if err == nil {
+		t.Errorf("files withoutSpdx Identifiers getting accepted")
+	}
+
+	// case 2 : The previous file (packaged or unpackaged does not contain spdxID)
+	tvPairs = append(tvPairs, reader.TagValuePair{Tag: "FileName", Value: "f2"})
+	_, err = ParseTagValues(tvPairs)
+	if err == nil {
+		t.Errorf("%s", err)
+	}
 
+	// case 3 : Invalid file with snippet
+	// Last unpackaged file before the packges start
+	// Last file of a package and New package starts
+	sid1 := spdx.ElementID("s1")
+	parser := tvParser2_1{
+		doc: &spdx.Document2_1{},
+		st:  psCreationInfo2_1,
+	}
+	fileName := "f2.txt"
+	_ = parser.parsePair2_1("FileName", fileName)
+	_ = parser.parsePair2_1("SnippetSPDXID", string(sid1))
+	err = parser.parsePair2_1("PackageName", "p2")
+	if err == nil {
+		t.Errorf("files withoutSpdx Identifiers getting accepted")
```

Comment:
Is this what you're actually testing? The line above with `SnippetSPDXID` means that we shifted from parsing a File section to parsing a Snippet section. That point is where the missing File SPDX ID was relevant. If you remove that line, then I think this test would be doing what you want it to do here.